identity_credential/src/validator/errors.rs

```diff
@@ -139,4 +140,65 @@ impl Display for CompoundPresentationValidationError {
   }
 }
 
-impl std::error::Error for CompoundPresentationValidationError {}
+impl Error for CompoundPresentationValidationError {}
+
+#[derive(Debug, thiserror::Error)]
+/// An error caused by a failure to verify a Domain Linkage configuration or credential.
+pub struct DomainLinkageVerificationError {
+  /// Cause of the error.
+  pub cause: DomainLinkageVerificationErrorCause,
+  /// Source of the error.
+  pub source: Option<Box<dyn Error + Send + Sync + 'static>>,
+}
+impl DomainLinkageVerificationError {
+  /// Provides the cause of the error.
+  pub fn get_cause(&self) -> &DomainLinkageVerificationErrorCause {
+    &self.cause
+  }
+}
+
+impl Display for DomainLinkageVerificationError {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(f, "{:?}; ", self.cause)?;
+    match &self.source {
+      Some(err) => write!(f, "{}", err)?,
+      None => {}
+    }
+    Ok(())
```

Comment:
```suggestion
    write!(f, "{}", self.cause)
```

Print the `Display` impl of the `cause`, not the `Debug` one.
Don't include the source error in the `Display` impl. Higher-level functionality/crates will do the formatting of these error chains. See also #1017.

Reply:
You can try `cargo clean`, that sometimes helps with IDE problems.